distributed/core.py

```diff
@@ -316,7 +342,39 @@ def __init__(
         connection_args=None,
         timeout=None,
         io_loop=None,
+        local_directory=None,
     ):
+        from distributed.nanny import Nanny
+
+        if local_directory is None:
+            local_directory = (
+                dask.config.get("temporary-directory") or tempfile.gettempdir()
```

Comment:
As is this only allows for setting a single `temporary-directory` config option and it needs to uniformly apply to the scheduler, worker, and nanny. I could see that being an issue if, for example, the worker and scheduler have sufficiently different filesystems. 

A workaround there would be to have separate config options for the scheduler and workers. Then parse the config option in the respective `<Class>.__init__(...)` method before forwarding it here. 

@jacobtomlinson do you have thoughts on how likely this situation is? Should we account for it here, or follow-up later if needed

Reply:
It's not uncommon for the client/scheduler to run on the user's machine (which is often Windows/Mac) and the workers on some remote linux system. Dask Jobqueue does this for example. @mrocklin was also talking recently about some PyTorch workflows that run the client and scheduler from a MacBook with GPU Linux servers as workers.

It's also pretty common for HPC systems to have some arbitrary tmp filesystem that folks are expected to use. 

I think it would be valuable to either have totally separate options, or keep this option and add an overriding option for the scheduler.

Reply:
Do we think this ought to be done here, or hammered out in a follow-up PR?

Reply:
I think a follow up would be fine but could you open an issue to track it?